python/mpt/src/mpt/trie_diff.cairo

```diff
@@ -289,19 +447,924 @@ func _process_storage_diff{
+    let subnode_left = OptionalUnionInternalNodeExtendedImpl.from_extended(subnode_left_extended);
+
+    let subnodes_right_ptr = cast(right.value.subnodes.value, felt*);
+    let subnode_right_extended = Extended(cast(subnodes_right_ptr[index], ExtendedEnum*));
+    %{ logger.debug_cairo("subnode_right_extended index: %s", ids.index) %}
+    let subnode_right = OptionalUnionInternalNodeExtendedImpl.from_extended(subnode_right_extended);
+
+    // path = path + bytes([i])
+    let path_copy = Bytes__copy__(path);
+    assert path_copy.value.data[path_copy.value.len] = index;
+    tempvar sub_path = Bytes(
+        new BytesStruct(data=path_copy.value.data, len=path_copy.value.len + 1)
+    );
+
+    return _compute_diff(
+        left=subnode_left, right=subnode_right, path=sub_path, account_address=account_address
+    );
+}
+
+// TODO: the none case could be split out.
+// / @notice Process differences in all branches of the right branch node
+// / @dev Recursively processes each branch with index 0-15
+// / @param left The Optional Leaf Node from the previous state
+// / @param subnodes The subnodes of the right branch node
+// / @param path The path traversed so far in the trie
+// / @param account_address The account address (if processing a storage trie)
+// / @param index The current branch index being processed (0-15)
+func _compute_left_leaf_diff_on_right_branch_node{
+    range_check_ptr,
+    bitwise_ptr: BitwiseBuiltin*,
+    poseidon_ptr: PoseidonBuiltin*,
+    node_store: NodeStore,
+    address_preimages: MappingBytes32Address,
+    storage_key_preimages: MappingBytes32Bytes32,
+    main_trie_end: AddressAccountNodeDictAccess*,
+    storage_trie_end: Bytes32U256DictAccess*,
+}(
+    left: OptionalUnionInternalNodeExtended,
+    subnodes: Subnodes,
+    path: Bytes,
+    account_address: Address,
+    index: felt,
+) -> () {
+    alloc_locals;
+
+    if (index == 16) {
+        return ();
+    }
+
+    // Use `branch_0` as the base pointer to the list of subnodes and index it as a felt*.
+    let subnodes_ptr = cast(subnodes.value, felt*);
+    let subnode_i_extended = Extended(cast(subnodes_ptr[index], ExtendedEnum*));
+    %{ logger.debug_cairo("subnode_i_extended index: %s", index) %}
+    let subnode_i = OptionalUnionInternalNodeExtendedImpl.from_extended(subnode_i_extended);
+
+    // TODO: optimization possible here
+
+    // path = path + bytes([i])
+    let path_copy = Bytes__copy__(path);
+    assert path_copy.value.data[path_copy.value.len] = index;
+    tempvar sub_path = Bytes(
+        new BytesStruct(data=path_copy.value.data, len=path_copy.value.len + 1)
+    );
+
+    // Three cases:
+    // 1. The first nibble of the subnode matches the branch node's key segment, in which
+    // case we compare to the leaf node shortened by the nibble,
+    // 2. It doesn't in which case we compare to None.
+    // 3. The leaf is None. (see 2.)
+
+    if (cast(left.value, felt) == 0) {
```

Comment:
can you even reach this since you catch null case before entering the function

Reply:
apologies, the name confused me, i thought in `_left_is_null` you would call smth like _compute_left_none_diff_on_right_branch_node